src/tracks/Pattern.cpp

```diff
@@ -65,6 +65,26 @@ Pattern::Pattern( InstrumentTrack * _instrument_track ) :
 	m_steps( MidiTime::stepsPerTact() )
 {
 	setName( _instrument_track->name() );
+
+	// Resize this track to be the same as existing tracks in the BB
+	const TrackContainer::TrackList & tracks =
+		m_instrumentTrack->trackContainer()->tracks();
+	for(unsigned int trackID = 0; trackID < tracks.size(); ++trackID)
+	{
+		if(tracks.at(trackID)->type() == Track::InstrumentTrack &&
```

Comment:
You should break on this condition. The reference track should be before the current track, not after. Then, you can remove the bounds check for `currentTCO`. It should look like:
````
if(type)
  if(not same track)
    update steps
  break
````